packages/template-retail-react-app/app/pages/account/profile.jsx

```diff
@@ -343,7 +349,7 @@ const PasswordCard = () => {
     )
 }
 
-const AccountDetail = () => {
+const AccountDetail = ({isSocialProfile = false}) => {
```

Comment:
As I get further down in this PR I see more and more prop drilling of this isSocialProfile. It's not a horrible pattern but no used as much these days. 

Did you ever thing about having this information made available via a provider/hook combo? Something that comes to mind is to have it implemented in the commerce-sdk-react so you can reuse some code there.

Reply:
@kevinxh would be a good person to ask about whether or not thats a good idea to implement in that lib.

Reply:
I think ideally the commerce-sdk-react library should handle the state `isSocialProfile`, it would be easy if that information is embedded in the jwt token, if not, we should seek to manually store the state in the Auth class as a class property.

something like

```js
// commerce-sdk-react auth.js

class Auth {
  isSocialLogin = false
  
  loginSocial = () => {
    // ...do the login
    this.isSocialLogin = true
    
    // potentially also store the IDP information like google, facebook, etc
  }
}

// Then we can provide a utility hook
// commerce-sdk-react useSocialLogin()
export const useSocialLogin = () => {
   const auth = useAuth()
   return auth.isSocialLogin
}
```

Reply:
@kevinxh If the info is embedded in the jwt token, would I then store it in local storage? Or is there another way for the template app to access this value?